system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp

```diff
@@ -0,0 +1,591 @@
+    } else {
+      const auto msg = "invalid state: " + std::to_string(mrm_state_.state);
+      throw std::runtime_error(msg);
+    }
+  }
+}
+
+autoware_adapi_v1_msgs::msg::MrmState::_behavior_type MrmHandler::getCurrentMrmBehavior()
+{
+  using autoware_adapi_v1_msgs::msg::MrmState;
+  using tier4_system_msgs::msg::OperationModeAvailability;
+
+  // State machine
+  if (mrm_state_.behavior == MrmState::NONE) {
+    if (operation_mode_availability_->pull_over) {
+      if (param_.use_pull_over) {
+        return MrmState::PULL_OVER;
+      }
+    }
+    if (operation_mode_availability_->comfortable_stop) {
+      if (param_.use_comfortable_stop) {
+        return MrmState::COMFORTABLE_STOP;
+      }
+    }
+    if (!operation_mode_availability_->emergency_stop) {
+      RCLCPP_WARN(this->get_logger(), "no mrm operation available: operate emergency_stop");
+    }
+    return MrmState::EMERGENCY_STOP;
+  }
+  if (mrm_state_.behavior == MrmState::PULL_OVER) {
+    if (operation_mode_availability_->pull_over) {
+      if (param_.use_pull_over) {
+        return MrmState::PULL_OVER;
+      }
+    }
+    if (operation_mode_availability_->comfortable_stop) {
+      if (param_.use_comfortable_stop) {
+        return MrmState::COMFORTABLE_STOP;
+      }
+    }
+    if (!operation_mode_availability_->emergency_stop) {
+      RCLCPP_WARN(this->get_logger(), "no mrm operation available: operate emergency_stop");
+    }
+    return MrmState::EMERGENCY_STOP;
+  }
+  if (mrm_state_.behavior == MrmState::COMFORTABLE_STOP) {
+    if (isStopped() && operation_mode_availability_->pull_over) {
+      if (param_.use_pull_over) {
+        return MrmState::PULL_OVER;
+      }
+    }
+    if (operation_mode_availability_->comfortable_stop) {
+      if (param_.use_comfortable_stop) {
+        return MrmState::COMFORTABLE_STOP;
+      }
+    }
+    if (!operation_mode_availability_->emergency_stop) {
+      RCLCPP_WARN(this->get_logger(), "no mrm operation available: operate emergency_stop");
+    }
+    return MrmState::EMERGENCY_STOP;
+  }
+  if (mrm_state_.behavior == MrmState::EMERGENCY_STOP) {
+    if (isStopped() && operation_mode_availability_->pull_over) {
+      if (param_.use_pull_over) {
+        return MrmState::PULL_OVER;
+      }
+    }
+    if (!operation_mode_availability_->emergency_stop) {
+      RCLCPP_WARN(this->get_logger(), "no mrm operation available: operate emergency_stop");
+    }
+    return MrmState::EMERGENCY_STOP;
+  }
+
+  return mrm_state_.behavior;
+}
+
+bool MrmHandler::isStopped()
+{
+  constexpr auto th_stopped_velocity = 0.001;
+  if (odom_->twist.twist.linear.x < th_stopped_velocity) {
+    return true;
+  }
+
+  return false;
+}
+
+bool MrmHandler::isEmergency() const
+{
+  return !operation_mode_availability_->autonomous || is_emergency_holding_;
+}
+
+bool MrmHandler::isArrivedAtGoal()
+{
+  using autoware_adapi_v1_msgs::msg::OperationModeState;
+
+  if (operation_mode_state_->mode == OperationModeState::STOP) {
```

Comment:
Why not just `return operation_mode_state_->mode == OperationModeState::STOP` ?

Reply:
just as you said！